pkg/controller/sparkapplication/controller.go

```diff
@@ -504,9 +499,7 @@ func (c *Controller) syncSparkApplication(key string) error {
 	case v1beta1.PendingSubmissionState:
 		// Check the status of the submission Job and set the application status accordingly.
 		succeeded, completionTime, err := c.subJobManager.hasJobSucceeded(appToUpdate)
-		if err != nil {
-			return err
```

Comment:
I prefer handling errors first.

Reply:
The method generates errors in two separate cases: when succeeded is nil (actual error trying to query job status) or when succeeded is false (error as to why the job failed). Hence this looks like a more cleaner way to handle it to me.

Reply:
Got it. Can you add some comments about the semantics here?

Reply:
done